llvm/tools/llvm-readobj/ELFDumper.cpp

```diff
@@ -5098,6 +5098,86 @@ template <class ELFT> void GNUELFDumper<ELFT>::printAddrsig() {
   }
 }
 
+template <class ELFT>
+static bool printAArch64PAuthABICoreInfo(raw_ostream &OS, uint32_t DataSize,
+                                         ArrayRef<uint8_t> Desc) {
+  OS << "    AArch64 PAuth ABI core info: ";
+  // DataSize - size without padding, Desc.size() - size with padding
+  if (DataSize != 16) {
+    OS << format("<corrupted size: expected 16, got %d>", DataSize);
+    return false;
+  }
+
+  uint64_t Platform =
+      support::endian::read64<ELFT::TargetEndianness>(Desc.data() + 0);
+  uint64_t Version =
+      support::endian::read64<ELFT::TargetEndianness>(Desc.data() + 8);
+
+  std::string PlatformDesc = [Platform]() {
```

Comment:
Consider `const StringRef` instead of a mutable std::string

Reply:
Thanks for suggestion. I've switched to `const char *` since we later use `PlatformDesc` as an argument for `format(...)` and expect it to be null-terminated. Storing null-terminated strings as `const char *` when we don't need anything except the pointer itself looks more clear in this context than calling `.data()` on `StringRef` and keeping in mind that in this particular case the `StringRef`'s content is also null-terminated (while it's not guaranteed in other cases).

See 8b562626454a9f6f9f9e41b8f84e17535924104d